pilot/pkg/model/push_context.go

```diff
@@ -880,7 +879,12 @@ func (ps *PushContext) GatewayServices(proxy *Proxy) []*Service {
 	for _, s := range svcs {
 		svcHost := string(s.Hostname)
 
-		if _, ok := hostsFromGateways[svcHost]; ok {
+		annotationMatches, _ := strconv.ParseBool(s.Attributes.Annotations["networking.istio.io/always-include-in-gateways"])
```

Comment:
Based on the previous discussion https://docs.google.com/document/d/1xHy2jQ8oiwMponMVY2zJr2eUAmHW_Hi9JK42a7cg5Pc/edit?tab=t.0#bookmark=id.z02np8mw7xzx there were concerns about such annotation - For which gateway's it should be added etc? I think those concerns are still relevant?

Reply:
Hmm, it would be hard to declare the name of a gateway since it applies to a Pod really. It seems likely you just have a couple of these you need to add and if the gateways get a couple extra clusters its not so bad?

Reply:
Yeah. The number may not be a concern. Is this mainly intended for using it in Envoy filter? Can you please give an example of Envoy filter that refers this cluster?

I think in general it might be useful for example a WASM filter making call but its interactions with exportTo etc might need to be documented well. The way we use/overcome this problem today is we create a gateway object with a localhost bind and a vs that attaches to that gateway - then we automatically get that cluster.